### ISSUE TITLE
[docdb] Intent metadata_write_time is incorrectly set

### ISSUE THREAD

**Issue by triump2020** [yugabyte/yugabyte-db#6909]
Hi ,  folllowing code block comes from [line222](https://github.com/yugabyte/yugabyte-db/blob/master/src/yb/tablet/transaction_participant.cc) ,  the 'data_copy ' seems that useless ?

`     auto data_copy = data;
      // We use hybrid time only for backward compatibility, actually wall time is required.
      data_copy.set_metadata_write_time(GetCurrentTimeMicros());`

**Comment by triump2020** [comment]
Above code block comes from : [line222](https://github.com/yugabyte/yugabyte-db/blob/master/src/yb/tablet/transaction_participant.cc#L222)

**Comment by tedyu** [comment]
I wonder if Sergei meant the following:
```
diff --git a/src/yb/tablet/transaction_participant.cc b/src/yb/tablet/transaction_participant.cc
index 6a37f42fa..4d5d63966 100644
--- a/src/yb/tablet/transaction_participant.cc
+++ b/src/yb/tablet/transaction_participant.cc
@@ -222,7 +222,7 @@ class TransactionParticipant::Impl
       auto data_copy = data;
       // We use hybrid time only for backward compatibility, actually wall time is required.
       data_copy.set_metadata_write_time(GetCurrentTimeMicros());
-      auto value = data.SerializeAsString();
+      auto value = data_copy.SerializeAsString();
       write_batch->Put(key.AsSlice(), value);
     }
     return true;
```

**Comment by triump2020** [comment]
Should be , but seems no impact on txn-correctness

**Comment by triump2020** [comment]
@bmatican   Currently , the hlc timestamp  of write intents host in  intent-db  is useless , since snapshot reads must wait for the safe time to past completely ,  right ?

**Comment by bmatican** [comment]
Thanks for digging into this! Seems like this is a bug, but indeed it should not affect txn correctness. I spoke to @spolitov and he mentioned that we would end up using the `metadata_write_time` when checking for aborted txns on compactions, but we would not actually be aborting anything that we shouldn't, so should be safe there.

The suggestion that @tedyu had above should be a good enough fix for this. @triump2020 would you mind putting a small PR for this? Figured to have the commit to your name, given you did the hard part of finding it.

**Comment by triump2020** [comment]
@bmatican   I ' m  very happy to do it  ,  though it 's my firt time !

**Comment by bmatican** [comment]
@triump2020 Hehe, even better :) always happy to take up new contributions!

Just submit a PR with the change, we'll run it internally though our test matrix and merge it if all looks good!

**Comment by triump2020** [comment]
@bmatican   Thanks , i 'll   do it   !

**Comment by triump2020** [comment]
@bmatican Please assign the task to me !  haha

**Comment by bmatican** [comment]
Fixed by 4c9b51408d066f4b08cedba64957f89b515361ac
